pytorch_lightning/core/memory.py

```diff
@@ -11,478 +11,24 @@
-        check=True,
-    )
-
-    # Convert lines into a dictionary
-    gpu_memory = [float(x) for x in result.stdout.strip().split(os.linesep)]
-    gpu_memory_map = {f"gpu_id: {gpu_id}/memory.used (MB)": memory for gpu_id, memory in enumerate(gpu_memory)}
-    return gpu_memory_map
-
-
-def get_formatted_model_size(total_model_size: float) -> float:
-    return f"{total_model_size:,.3f}"
-
-
-def get_human_readable_count(number: int) -> str:
-    """
-    Abbreviates an integer number with K, M, B, T for thousands, millions,
-    billions and trillions, respectively.
-
-    Examples:
-        >>> get_human_readable_count(123)
-        '123  '
-        >>> get_human_readable_count(1234)  # (one thousand)
-        '1.2 K'
-        >>> get_human_readable_count(2e6)   # (two million)
-        '2.0 M'
-        >>> get_human_readable_count(3e9)   # (three billion)
-        '3.0 B'
-        >>> get_human_readable_count(4e14)  # (four hundred trillion)
-        '400 T'
-        >>> get_human_readable_count(5e15)  # (more than trillion)
-        '5,000 T'
-
-    Args:
-        number: a positive integer number
-
-    Return:
-        A string formatted according to the pattern described above.
-
-    """
-    assert number >= 0
-    labels = PARAMETER_NUM_UNITS
-    num_digits = int(np.floor(np.log10(number)) + 1 if number > 0 else 1)
-    num_groups = int(np.ceil(num_digits / 3))
-    num_groups = min(num_groups, len(labels))  # don't abbreviate beyond trillions
-    shift = -3 * (num_groups - 1)
-    number = number * (10 ** shift)
-    index = num_groups - 1
-    if index < 1 or number >= 100:
-        return f"{int(number):,d} {labels[index]}"
-
-    return f"{number:,.1f} {labels[index]}"
-
-
-def _is_lazy_weight_tensor(p: Tensor) -> bool:
-    if _TORCH_GREATER_EQUAL_1_8:
-        from torch.nn.parameter import UninitializedParameter
-
-        if isinstance(p, UninitializedParameter):
-            warning_cache.warn(
-                "A layer with UninitializedParameter was found. "
-                "Thus, the total number of parameters detected may be inaccurate."
-            )
-            return True
-    return False
+from pytorch_lightning.utilities import rank_zero_deprecation
+
+rank_zero_deprecation(
+    "pytorch_lightning.core.memory.get_memory_profile and "
+    "pytorch_lightning.core.memory.get_gpu_memory_map have been moved "
+    "to pytorch_lightning.utilities.memory since v1.5 and will be "
+    "removed in v1.7."
+)
+
+# To support backward compatibility as get_memory_profile and get_gpu_memory_map have been moved
+from pytorch_lightning.utilities.memory import get_gpu_memory_map, get_memory_profile  # noqa: E402 F401 # isort: skip
+
+rank_zero_deprecation(
+    "pytorch_lightning.core.memory.LayerSummary and "
+    "pytorch_lightning.core.memory.ModelSummary have been moved "
+    "to pytorch_lightning.utilities.model_summary since v1.5 and "
+    "will be removed in v1.7."
```

Comment:
```suggestion
    "`pytorch_lightning.core.memory.LayerSummary` and"
    " `pytorch_lightning.core.memory.ModelSummary` have been moved"
    " to `pytorch_lightning.utilities.model_summary` since v1.5 and will be removed in v1.7."
```